homeassistant/components/binary_sensor/hive.py

```diff
@@ -35,9 +35,24 @@ def __init__(self, hivesession, hivedevice):
         self.attributes = {}
         self.data_updatesource = '{}.{}'.format(self.device_type,
                                                 self.node_id)
-
+        self.uniqueid = '{}-{}'.format(self.node_id, self.device_type)
         self.session.entities.append(self)
 
+    @property
+    def unique_id(self):
+        """Return unique ID of entity."""
+        return self.uniqueid
+
+    @property
+    def device_info(self):
```

Comment:
This requires implementation of config entries first. Which should be done in a  separate PR. After that is done another PR can be made, adding `device_info`.

https://developers.home-assistant.io/docs/en/config_entries_index.html

https://developers.home-assistant.io/docs/en/device_registry_index.html#defining-devices

Reply:
This code change has been moved to a new PR